src/Init/Prelude.lean

```diff
@@ -941,50 +942,142 @@ be eagerly evaluated (see `ite`).
   | true  => x
   | false => y
 
+/--
+`or x y`, or `x || y`, is the boolean "or" operation (not to be confused
+with `Or : Prop → Prop → Prop`, which is the propositional connective).
+It is `@[macroInline]` because it has C-like short-circuiting behavior:
+if `x` is true then `y` is not evaluated.
+-/
 @[macroInline] def or (x y : Bool) : Bool :=
   match x with
   | true  => true
   | false => y
 
+/--
+`and x y`, or `x && y`, is the boolean "and" operation (not to be confused
+with `And : Prop → Prop → Prop`, which is the propositional connective).
+It is `@[macroInline]` because it has C-like short-circuiting behavior:
+if `x` is false then `y` is not evaluated.
+-/
 @[macroInline] def and (x y : Bool) : Bool :=
   match x with
   | false => false
   | true  => y
 
+/--
+`not x`, or `!x`, is the boolean "not" operation (not to be confused
+with `Not : Prop → Prop`, which is the propositional connective).
+-/
 @[inline] def not : Bool → Bool
   | true  => false
   | false => true
 
-/-- The type of natural numbers. `0`, `1`, `2`, ...-/
+/--
+The type of natural numbers, starting at zero. It is defined as an
+inductive type freely generated by "zero is a natural number" and
+"the successor of a natural number is a natural number".
+
+You can prove a theorem `P n` about `n : Nat` by `induction n`, which will
```

Comment:
The docs here are to some extent supposed to be tutorial-like; the intent is for them to function like Kevin's docs on "how do I make this? how do I use this?". Plus, Nat is pretty intrinsically connected to induction - people will want to know how to do induction on Nat specifically even though it's a general mechanism. The special casing information is later because it is more advanced.